src/components/Stake/CardAccordion.js

```diff
@@ -0,0 +1,173 @@
+
+  const toggleButton = useCallback((index) => {
+    setOpened((opened) => !opened)
+  }, [])
+
+  return (
+    <div
+      css={`
+        position: relative;
+        margin-bottom: ${RADIUS}px;
+      `}
+    >
+      <OpenedSurfaceBorder opened={opened} />
+      <Card
+        css={`
+          width: 100%;
+          z-index: 2;
+        `}
+      >
+        <ToggleButton onClick={toggleButton} opened={opened} />
+        {card}
+      </Card>
+      <div
+        css={`
+          margin-top: ${-RADIUS}px;
+        `}
+      >
+        <Transition
+          native
+          items={opened}
+          from={{ height: 0 }}
+          enter={{ height: 'auto' }}
+          leave={{ height: 0 }}
+        >
+          {(show) =>
+            show &&
+            ((props) => (
+              <Expansion
+                css={`
+                  background: ${theme.surfaceUnder};
+                  overflow: hidden;
+                `}
+                style={props}
+              >
+                <div
+                  css={`
+                    width: 100%;
+                    overflow: hidden;
+
+                    padding: ${4 * GU}px ${3 * GU}px ${3 * GU}px;
+                    box-shadow: inset 0 ${RADIUS + 4}px 4px -4px rgba(0, 0, 0, 0.16);
+                  `}
+                >
+                  {expansion}
+                </div>
+              </Expansion>
+            ))
+          }
+        </Transition>
+      </div>
+    </div>
+  )
+})
+
+const Expansion = styled(animated.div)`
+  z-index: 1;
+`
+
+CardAccordion.propTypes = {
+  card: PropTypes.node,
+  expansion: PropTypes.node,
+}
+
+function ToggleButton({ onClick, opened }) {
+  const theme = useTheme()
+  return (
+    <ButtonIcon
+      label={opened ? 'Close' : 'Open'}
+      focusRingRadius={RADIUS}
+      onClick={onClick}
+      css={`
+        position: absolute;
+        top: ${3.5 * GU}px;
+        left: ${3.5 * GU}px;
+        display: flex;
+        flex-direction: column;
+        color: ${theme.surfaceContentSecondary};
+        & > div {
+          display: flex;
+          transform-origin: 50% 50%;
+          transition: transform 250ms ease-in-out;
+        }
+      `}
+    >
+      <div
+        css={`
+          transform: rotate3d(${opened ? 1 : 0}, 0, 0, 180deg);
+          transform: rotate3d(0, 0, ${opened ? 1 : 0}, 180deg);
+        `}
+      >
+        <IconUp size="small" />
+      </div>
+      <div
+        css={`
+          transform: rotate3d(${opened ? -1 : 0}, 0, 0, 180deg);
+          transform: rotate3d(0, 0, ${opened ? -1 : 0}, 180deg);
+        `}
+      >
+        <IconDown size="small" />
+      </div>
+    </ButtonIcon>
+  )
+}
+
+ToggleButton.propTypes = {
+  opened: PropTypes.bool,
+  onClick: PropTypes.func,
+}
+
+function OpenedSurfaceBorder({ opened }) {
+  return (
+    <Spring
+      native
+      from={{ width: 0 }}
+      to={{ width: Number(opened) }}
+      config={{ mass: 0.8, tension: 300, friction: 28 }}
```

Comment:
Is there anything in our existing springs that would work nicely here? you can check in `styles/springs`
No worries if not, but we could move it there too if you think it feels unique and could be useful in other places!